src/main/java/org/candlepin/subscriptions/resource/InstancesResource.java

```
@@ -216,6 +226,25 @@ private InstanceData asTallyHostViewApiInstance(
     return instance;
   }
 
+  private static List<HardwareMeasurementType> getHardwareMeasurementTypesFromCategory(
+      ReportCategory reportCategory) {
+    if (Objects.isNull(reportCategory)) {
+      return new ArrayList<>();
+    }
+    switch (reportCategory) {
+      case VIRTUAL:
+        return List.of(HardwareMeasurementType.VIRTUAL);
+      case PHYSICAL:
+        return List.of(HardwareMeasurementType.PHYSICAL);
+      case HYPERVISOR:
+        return List.of(HardwareMeasurementType.HYPERVISOR);
+      case CLOUD:
+        return HardwareMeasurementType.getCloudProviderTypes();
+      default:
+        return new ArrayList<>();
```

Would rather raise an exception here.